### DISCUSSION TITLE
Custom Component

### DISCUSSION THREAD

**Discussion by yosephberhanu** [Budibase/budibase#5270] [Ideas]
Hi all, 
I've been looking into budibase as a potential platform for our SaaS products. We currently use micro-services and would like to migrate to budibase to improve delivery time. However the biggest challenge has been our inability to build custom components within budibase which we can reuse. The components I'm talking about include things such as open layers based maps for data entry and such. Is there a possibility to do this at the moment within budibase that I overlooked or if it does not exist is there a plan to include it in the near feature. Thank you everyone for the amazing work on budibase and the responses in advance.

Regards,

**Comment by melohagan** [comment]
Hi @yosephberhanu 

A map component has recently been added which may be of use: https://docs.budibase.com/docs/map

Custom components are on the roadmap for Q4: https://github.com/Budibase/budibase/projects/10

**Reply by yosephberhanu** [reply]
Thank you @shogunpurple  for the quick reply. The kind of maps we are working on require a bit more interactions (creating, editing polygons, splitting, line drawing and so on). Maybe the custom components will address our requirements. We can and would like to contribute in that direction and we hope to get started on that soon now that we know there is a plan to have custom components. Thank you again.
